pkg/api/registered/registered.go

```diff
@@ -25,71 +26,97 @@ import (
 	"k8s.io/kubernetes/pkg/api/unversioned"
 )
 
-// List of registered API versions.
-// The list is in the order of most preferred to the least.
-var RegisteredGroupVersions []unversioned.GroupVersion
+var (
+	// registeredVersions represents all enabled API versions. Please call
+	// RegisterVersions() to register the versions.
+	registeredVersions = map[unversioned.GroupVersion]struct{}{}
```

Comment:
In the API Machinery call, we discussed splitting into `registeredVersion`, which is the list of all available group/versions and `enabledVersions`, which seems to be how this is godoced.

Can you make that split here in the variables even if you just update the `enabledVersions` list at the same time.  I think it will help us reason about the stages.